src/nacl/bindings/crypto_secretstream.py

```diff
@@ -317,10 +327,16 @@ def crypto_secretstream_xchacha20poly1305_pull(state, c, ad=None):
     )
     ensure(rc == 0, "Unexpected failure", raising=exc.RuntimeError)
 
-    return (ffi.buffer(state.rawbuf, mlen)[:], int(state.tagbuf[0]))
+    # Cast safety: we `ensure` above that `state.tagbuf is not None`.
+    return (
+        ffi.buffer(state.rawbuf, mlen)[:],
```

Comment:
Sorry, comment went on some random commit rather than this PR before. Is there an advantage to a cast here rather than just asserting not none so myth understands the type narrowing?

Reply:
No strong opinions here! I don't think there's advantage as such; I was just reluctant to disrupt the pattern of using `ensures` across the library. This seemed like a less disruptive change.

More generally, I suppose casts can
- narrow when working with types that only exist at typecheck time, (e.g. `Protocol`s that aren't marked as `runtime_checkable`
-- avoid runtime checks that may occur if asserts are enabled.
 
E.g. I could `cast(Dict[str, int], my_dict)` instead of doing
```python
for k, v in my_dict.items():
    assert isinstance(k, str)
    assert isinstance(v, int)
```

(I'm not sure if mypy is able to narrow based on that loop though.)